mlir/utils/performance/analysis/compileAndCollectTuningData.py

```diff
@@ -0,0 +1,604 @@
+                        'sgpr_spills': result_dict.get('sgpr_spills', ''),
+                        'LDS_allocated': result_dict.get('LDS_allocated', ''),
+                        'occupancy': result_dict.get('occupancy', ''),
+                        'WFsPerWG': result_dict.get('WFsPerWG', ''),
+                        'mfma_wmma_instruction': result_dict.get('mfma_wmma_instruction', '')
+                    }
+                
+                writer.writerow(row)
+        
+        print(f"\nResults written to {output_file}")
+        
+    except Exception as e:
+        print(f"\nError writing results to tsv: {e}")
+
+def print_progress(current, total):
+    """Print a progress bar to stdout."""
+    prefix = "Processing Configs"
+    percent = (current / total) * 100
+    bar_length = 40
+    filled_length = int(bar_length * current // total)
+    bar = '█' * filled_length + '-' * (bar_length - filled_length)
+    print(f'\r{prefix}: |{bar}| {current}/{total} ({percent:.1f}%)', end='',
+          flush=True)
+    if current == total:
+        print()  # New line when complete
+
+def main():
+    """Main function to process configurations and collect tuning data."""
+    parser = argparse.ArgumentParser(
+        description="Compile configurations and collect tuning data",
+        formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+    parser.add_argument('--op', required=True,
+                        help='Operation to perform (e.g., "compile")',
+                        choices=['conv', 'gemm', 'attention'])
+    parser.add_argument('config_tsv', help='Path to the tuning database file')
+    
+    args = parser.parse_args()
+
+    # Get the paths to the rocmlir binaries
+    build_bin_dir = os.path.dirname(os.path.abspath(__file__))
+    rocmlir_root = os.path.dirname(build_bin_dir)
+    paths = perfRunner.create_paths(None, rocmlir_root)
+
+    # Parse the configuration file
+    configs = perfRunner.read_tuning_db(args.config_tsv, True)
+    op_configs = None
+    if (args.op == 'conv'):
+        op_configs = perfRunner.getConvConfigurations(args.config_tsv)
+    elif (args.op == 'gemm'):
+        op_configs = perfRunner.getGemmConfigurations(args.config_tsv)
+    elif (args.op == 'attention'):
+        op_configs = perfRunner.getAttentionConfigurations(args.config_tsv)
+    else:
+        print(f"Error: Unknown operation '{args.op}'")
+        sys.exit(1)
+    print(f"Found {len(configs)} configurations to process")
+    
+    # Process each configuration
+    results = []
+    total_configs = len(configs)
+    for i, (config, test_args) in enumerate(zip(configs, op_configs)):
+        print_progress(i, total_configs)
+        metrics = compile_and_collect_data(config, configs[config], test_args,
+                                           args.op, paths)
+        results.append(metrics)
+
+    #Write the results to a final tsv file
+    write_results_to_tsv(results, configs)
+
+if __name__ == "__main__":
+    main()
```

Comment:
If we plan to use this script as part of CI (since it is added to ci-performance-scripts) it would be good to pass main's return value so that we can ensure that CI would fail if we had non-zero return value at some point. (we had a lot of problems previously with different python scripts which weren't propagating errors properly, which led to silent failures in CI)

```suggestion
    sys.exit(main())
```